packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+          }
+
+          fileSystem.file(parsedArgs['output'])
+            ..createSync()
+            ..writeAsStringSync(outputContents.toString());
+        },
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_lowercase_transformer',
+          args: <String>[],
+        ),
+        const AssetTransformerEntry(
+          package: 'my_distance_from_ascii_a_transformer',
+          args: <String>[],
+        ),
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+    expect(fileSystem.file(outputPath).readAsStringSync(), '012');
+  });
+
+  testWithoutContext('prints an error when a transformer in a chain (thats not the first) does not produce an output', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')
+      ..createSync()
+      ..writeAsStringSync('ABC');
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_lowercase_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+              ..addOption('input')
+              ..addOption('output'))
+            .parse(args);
+
+          final String inputFileContents = fileSystem.file(parsedArgs['input']).readAsStringSync();
+          fileSystem.file(parsedArgs['output'])
+            ..createSync()
+            ..writeAsStringSync(inputFileContents.toLowerCase());
+        },
+      ),
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_distance_from_ascii_a_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          // Do nothing.
+        },
+        stderr: 'Transformation failed, but I forgot to exit with a non-zero code.'
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
```

Comment:
let's expect on the return value

Reply:
Addressed in [expect on transformationResult](https://github.com/flutter/flutter/pull/143815/commits/7a9708ec4d120bca184689574016098ad0164fcd)